helix-view/src/handlers/dap.rs

```diff
@@ -274,6 +275,65 @@ impl Editor {
                         self.set_status("Debugged application started");
                     }; // TODO: do we need to handle error?
                 }
+                Event::Terminated(terminated) => {
+                    let restart_args = if let Some(terminated) = terminated {
+                        terminated.restart
+                    } else {
+                        None
+                    };
+                    let restart = restart_args.is_some();
+                    let disconnect_args = Some(DisconnectArguments {
+                        restart: Some(restart),
+                        terminate_debuggee: None,
+                        suspend_debuggee: None,
+                    });
+
+                    let starting_request = debugger.starting_request();
+                    let result = debugger.disconnect(disconnect_args).await;
+                    if result.is_ok() {
```

Comment:
Rewrite this so it's 

```
if let Err(ere) = result {
   set_error(...)
   return;
}
```

That way the error exits early and the regular path isn't indented inside an if